File: src/main/kotlin/a/QueueAtTheSchool.kt
package a

/* Problem Link: https://codeforces.com/problemset/problem/266/B */

fun main() {
    val (n, t) = readln().split(" ").map { it.toInt() }
    val queue = readln().toCharArray()

    for (i in 1..t) {
        var j = 0
        while (j < queue.size - 1) {
            if(queue[j] == 'B' && queue[j + 1] == 'G') {
                val temp = queue[j]
                queue[j] = queue[j + 1]
                queue[j + 1] = temp
                j++
            }

            j++
        }
    }

    print(queue)
}

File: Codeforces/C++/problemset/Queue at the School.cpp
// Problem Link: https://codeforces.com/problemset/problem/266/B

#include <iostream>
using namespace std;
typedef long long ll;
int main()
{
    ll n,t;
    cin >> n >> t;
    char *q = new char[n];
    for(ll i = 0; i < n; i++)
        cin >> q[i];
    for(ll i = 0; i < t; i++)
    {
        for(ll j = 0; j < n-1; j++)
        {
            if(q[j] == 'B'&& q[j+1] == 'G')
            {
                swap(q[j],q[j+1]);
                j++;
            }
        }
    }
    for(ll i = 0; i < n; i++)
        cout << q[i];

    delete [] q;
    return 0;
}
